### PR TITLE
Refactoring pose and vicon code

### PR DESCRIPTION
**Pull request by simbamford**
Combined vicon examples with example of constructing a pose dict fro scratch; refactored importVicon to use library functions for splitting by a label; renamed importVicon to importIitVicon; Put plot_trajectories function together with PlotPose function.

### CODE REVIEW

**Comment by arrenglover** [comment]
ping @ghoshsuman. we should try to look at pull requests within a day if we can - at least to comment that you can't do it if that is the case.

**Comment by ghoshsuman** [comment]
I actually started the review on Friday, but got sidetracked by other stuff. I'll take care of it, no worries.

File: python/libraries/bimvee/geometry.py [hunk 1]
```diff
@@ -114,6 +147,28 @@ def combineTwoQuaternions(q1, q2):
     qOut[1:4] = vOut
     return qOut
 
+'''
+Expects 
+    - poseDict in bimvee form {'ts', 'point', 'rotation' as above}
+    - translation as np array of x,y,z
+    - rotation as a np array of w,x,y,z (quaternion)
```
Comment (ghoshsuman) [inline python/libraries/bimvee/geometry.py:154]: might be useful to explicitly mention that the rotations are defined wrt local axes unlike the translations

File: python/libraries/bimvee/importIitVicon.py [hunk 2]
```diff
@@ -39,7 +40,7 @@
                     'point' : <2D array where each row has 3d position of a body at a time instant>,
                     'rotation' : <2D array where each row has rotation of a body at a time instant expressed as a quaternion (4d)>,
                     'bodyId' : <1D array where each row has the bodyId of the corresponding marker>,
```
Comment (ghoshsuman) [inline python/libraries/bimvee/importIitVicon.py:42]: weren't we going to refactor this name to 'label' ?

File: python/libraries/bimvee/importIitVicon.py [hunk 3]
```diff
@@ -56,22 +57,25 @@
                     'point' : <2D array where each row has 3d position of a body at a time instant>,
                     'rotation' : <2D array where each row has the rotation of a body at a time instant expressed as a quaternion (4d)>,
                     'bodyId' : <1D array where each row has the bodyId of the corresponding marker>,
```
Comment (ghoshsuman) [inline python/libraries/bimvee/importIitVicon.py:59]: weren't we going to refactor this to something like 'label'?

File: python/libraries/bimvee/importIitVicon.py [hunk 4]
```diff
@@ -129,45 +136,31 @@ def importVicon(**kwargs):
...
-            zeroTimestampsForAChannel(outDict['data']['vicon']) # TODO: Zeroing in the separated channel case
+    poseDict['ts'] = np.array(poseDict['ts'], dtype=np.float64)
+    poseDict['point'] = np.array(poseDict['point'], dtype=np.float64)
+    poseDict['rotation'] = np.array(poseDict['rotation'], dtype=np.float64)
+    poseDict['bodyId'] = np.array(poseDict['bodyId'], dtype=object)
+    if kwargs.get('zeroTime', kwargs.get('zeroTimestamps', True)):
+        zeroTimestampsForADataType(poseDict)
+    if kwargs.get('separateBodiesAsChannels', False):
+        outDict['info']['uniqueIds'] = np.unique(poseDict['bodyId'])
+        separatedBodies = splitByLabel(poseDict, 'bodyId')
```
Comment (ghoshsuman) [inline python/libraries/bimvee/importIitVicon.py:154]: Does splitByLabel() also delete the 'bodyId' key-value pair inside poseDict?

File: python/libraries/bimvee/plotPose.py [hunk 5]
```diff
@@ -63,4 +66,35 @@ def plotPose(inDict, **kwargs):
...
+    ax = kwargs.get('ax')
+    if ax is None:
+        ax = plt.axes(projection='3d')
+        kwargs['ax'] = ax
+    for name in bodyIds:
+        select_body = all([(inc in name) for inc in include]) and all([not (exc in name) for exc in exclude])
+        if select_body:  # modify this line to plot whichever markers you want
+            marker_pose = viconDataDict['data'][name]['pose6q']['point']
+            ax.scatter3D(marker_pose[:, 0], marker_pose[:, 1], marker_pose[:, 2], label=name)
+    ax.set_xlabel('X Label')
```
Comment (ghoshsuman) [inline python/libraries/bimvee/plotPose.py:89]:
```suggestion
    ax.set_xlabel('X')
```

File: python/libraries/bimvee/plotPose.py [hunk 6]
```diff
@@ -63,4 +66,35 @@ def plotPose(inDict, **kwargs):
...
+    ax.set_ylabel('Y Label')
```
Comment (ghoshsuman) [inline python/libraries/bimvee/plotPose.py:90]:
```suggestion
    ax.set_ylabel('Y')
```

File: python/libraries/bimvee/plotPose.py [hunk 7]
```diff
@@ -63,4 +66,35 @@ def plotPose(inDict, **kwargs):
...
+    ax.set_zlabel('Z Label')
```
Comment (ghoshsuman) [inline python/libraries/bimvee/plotPose.py:91]:
```suggestion
    ax.set_zlabel('Z')
```

File: python/libraries/bimvee/importIitVicon.py (hunk 4 above)
Comment (simbamford) [reply python/libraries/bimvee/importIitVicon.py:154]: No - it leaves it

**Pull request merged**